test/src/sdk-harness/test_projects/storage_map/src/main.sw

```diff
@@ -0,0 +1,281 @@
+
+    fn insert_into_u64_to_tuple_map(key: u64, value: (b256, u8, bool));
+    fn get_from_u64_to_tuple_map(key: u64) -> (b256, u8, bool);
+
+    fn insert_into_u64_to_struct_map(key: u64, value: Struct);
+    fn get_from_u64_to_struct_map(key: u64) -> Struct;
+
+    fn insert_into_u64_to_enum_map(key: u64, value: Enum);
+    fn get_from_u64_to_enum_map(key: u64) -> Enum;
+
+    fn insert_into_u64_to_str_map(key: u64, value: str[33]);
+    fn get_from_u64_to_str_map(key: u64) -> str[33];
+
+    fn insert_into_u64_to_array_map(key: u64, value: [b256;
+    3]);
+    fn get_from_u64_to_array_map(key: u64) -> [b256;
+    3];
+
+    fn insert_into_bool_to_u64_map(key: bool, value: u64);
+    fn get_from_bool_to_u64_map(key: bool) -> u64;
+
+    fn insert_into_u8_to_u64_map(key: u8, value: u64);
+    fn get_from_u8_to_u64_map(key: u8) -> u64;
+
+    fn insert_into_u16_to_u64_map(key: u16, value: u64);
+    fn get_from_u16_to_u64_map(key: u16) -> u64;
+
+    fn insert_into_u32_to_u64_map(key: u32, value: u64);
+    fn get_from_u32_to_u64_map(key: u32) -> u64;
+
+    fn insert_into_tuple_to_u64_map(key: (b256, u8, bool), value: u64);
+    fn get_from_tuple_to_u64_map(key: (b256, u8, bool)) -> u64;
+
+    fn insert_into_struct_to_u64_map(key: Struct, value: u64);
+    fn get_from_struct_to_u64_map(key: Struct) -> u64;
+
+    fn insert_into_enum_to_u64_map(key: Enum, value: u64);
+    fn get_from_enum_to_u64_map(key: Enum) -> u64;
+
+    fn insert_into_str_to_u64_map(key: str[33], value: u64);
+    fn get_from_str_to_u64_map(key: str[33]) -> u64;
+
+    fn insert_into_array_to_u64_map(key: [b256;
+    3], value: u64);
+    fn get_from_array_to_u64_map(key: [b256;
+    3]) -> u64;
+}
+
+impl StorageMapTest for Contract {
+    fn init() {
+        storage.map1 = ~StorageMap::new::<u64, bool>();
+        storage.map2 = ~StorageMap::new::<u64, u8>();
+        storage.map3 = ~StorageMap::new::<u64, u16>();
+        storage.map4 = ~StorageMap::new::<u64, u32>();
+        storage.map5 = ~StorageMap::new::<u64, u64>();
+        storage.map6 = ~StorageMap::new::<u64, (b256, u8, bool) >();
+        storage.map7 = ~StorageMap::new::<u64, Struct>();
+        storage.map8 = ~StorageMap::new::<u64, Enum>();
+        storage.map9 = ~StorageMap::new::<u64, str[33]>();
+        storage.map10 = ~StorageMap::new::<u64, [b256;
+        3]>();
+
+        storage.map11 = ~StorageMap::new::<bool, u64>();
+        storage.map12 = ~StorageMap::new::<u8, u64>();
+        storage.map13 = ~StorageMap::new::<u16, u64>();
+        storage.map14 = ~StorageMap::new::<u32, u64>();
+        storage.map15 = ~StorageMap::new::<(b256, u8, bool), u64 >();
+        storage.map16 = ~StorageMap::new::<Struct, u64>();
+        storage.map17 = ~StorageMap::new::<Enum, u64>();
+        storage.map18 = ~StorageMap::new::<str[33], u64>();
+        storage.map19 = ~StorageMap::new::<[b256;
+        3], u64>();
+    }
+
+    fn insert_into_u64_to_bool_map(key: u64, value: bool) {
```

Comment:
what about just: `fn insert_into_map1(...`

The function signature gives insight into the type of mapping this is.

Reply:
I thought it would be easier to know how to call these from the test if they have the types in the signature instead of having to look up what `map1` is and so on, but we can bikeshed over this!

Reply:
agreed that it's easier for the tester the way you have it.